src/components/Bookmark.tsx

```diff
@@ -0,0 +1,77 @@
+import BookmarkBorderOutlinedIcon from "@mui/icons-material/BookmarkBorderOutlined";
+import BookmarkOutlinedIcon from "@mui/icons-material/BookmarkOutlined";
+import { Button } from "@mui/material";
+import { useState } from "react";
+import { useBookmarkRecipe } from "../services/mutations/useBookmarkRecipe";
+import { useDeleteBookmarkedRecipe } from "../services/mutations/useDeleteBookmarkedRecipe";
+
+type BookmarkProps = {
+  withText?: boolean;
+  recipeId: string;
+};
+
+const withTextStyle = {
+  gap: 0.5,
+};
+
+const withoutTextStyle = {
+  padding: 1,
+  minWidth: 0,
+  boxShadow: "none",
+  borderColor: "background.lightGrey",
+  borderWidth: 1.5,
+  backgroundColor: "background.default",
+  color: "primary.main",
+  "&:hover": {
+    backgroundColor: "background.lightGrey",
+    boxShadow: "none",
+  },
+};
+
+const Bookmark = ({ recipeId, withText = false }: BookmarkProps) => {
+  const [bookmarked, setBookmarked] = useState<boolean>(false);
+  const [bookmarkId, setBookmarkId] = useState<string>("");
+  const { isSaving, bookmarkRecipe } = useBookmarkRecipe();
+  const { isDeleting, deleteBookmarkedRecipe } = useDeleteBookmarkedRecipe();
+
+  //will change after implementing authentication functionality
+  const isAuthenticated = true;
+  const userId = "UgmRjQ854xMU3ST8LiVv";
+
+  const handleClick = () => {
```

Comment:
it's not necessarily important to do that, but I like adding types to functions, it's clear what to expect from it:
const handleClick = (): void => ...
There are different opinions about it. Some of programmers writes it and adds rules to eslinter, some don't. so you can decide if the return type of function should be written or not :)